applications/CSharpWrapperApplication/tests/cpp_tests/test_calculate.cpp

```diff
@@ -0,0 +1,139 @@
+//    _____  _____ _                  __          __                                                 _ _           _   _
+//   / ____|/ ____| |                 \ \        / /                               /\               | (_)         | | (_)
+//  | |    | (___ | |__   __ _ _ __ _ _\ \  /\  / / __ __ _ _ __  _ __   ___ _ __ /  \   _ __  _ __ | |_  ___ __ _| |_ _  ___  _ __
+//  | |     \___ \| '_ \ / _` | '__| '_ \ \/  \/ / '__/ _` | '_ \| '_ \ / _ \ '__/ /\ \ | '_ \| '_ \| | |/ __/ _` | __| |/ _ \| '_  |
+//  | |____ ____) | | | | (_| | |  | |_) \  /\  /| | | (_| | |_) | |_) |  __/ | / ____ \| |_) | |_) | | | (_| (_| | |_| | (_) | | | |
+//   \_____|_____/|_| |_|\__,_|_|  | .__/ \/  \/ |_|  \__,_| .__/| .__/ \___|_|/_/    \_\ .__/| .__/|_|_|\___\__,_|\__|_|\___/|_| |_|
+//                                 | |                     | |   | |                    | |   | |
+//                                 |_|                     |_|   |_|                    |_|   |_|
+//
+//
+//  License: BSD License
+//   license: CSharpWrapperApplication/license.txt
+//
+//  Main authors:  Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "containers/model.h"
+#include "external_interface.h"
+#include "utilities/os_utilities.h"
+
+namespace Kratos
+{
+    namespace Testing
+    {
+        void CreateMDPAFile()
+        {
+            std::filebuf buffer;
+            buffer.open(OSUtilities::GetCurrentWorkingDir() + "/file.mdpa",std::ios::out);
+            std::ostream os(&buffer);
+            os << "Begin ModelPartData\nEnd ModelPartData\n\nBegin Properties  0\n    DENSITY 2700.000000\n    YOUNG_MODULUS 7000000.000000\n    POISSON_RATIO 0.300000\n    BODY_FORCE [3] (0.000000,0.000000,0.000000)\n    THICKNESS 1.000000\nEnd Properties\n\nBegin Nodes\n        1        0.0        0.0         0.0\n        2        0.0        0.0         1.0\n        3        1.0        0.0         0.0\n        4        1.0        1.0         0.0\nEnd Nodes\n\nBegin Elements SmallDisplacementElement3D4N\n    1 0 1 2 3 4\nEnd Elements\n\nBegin SubModelPart BasePart // Note that this would be a sub sub modelpart\n    Begin SubModelPartNodes\n        1\n        2\n    End SubModelPartNodes\n    Begin SubModelPart inner_part\n        Begin SubModelPartNodes\n            1\n        End SubModelPartNodes\n    End SubModelPart\nEnd SubModelPart";
+            buffer.close();
+        }
+
+        /**
+        * Checks the correct work of update node position
+        */
+        KRATOS_TEST_CASE_IN_SUITE(CSharpWrapperUpdateNodePosition, KratosCSharpWrapperApplicationFastSuite)
+        {
+            // In case the StructuralMechanicsApplciation is not compiled we skip the test
+            if (!KratosComponents<Element>::Has("SmallDisplacementElement3D4N"))
```

Comment:
why not
```suggestion
            if (!KratosComponents<Element>::Has("Element2D4N"))
```
?
then there would not be a dependency on StructMech

Reply:
If you look the code of the app in detail you will see that right now is hardcoded to depend on StructMech

Reply:
Not being such hardcoded is one of the future works of @Hubertus248